students/HABTAMU/lesson09/assignment/charges_calc.py

```diff
@@ -0,0 +1,134 @@
+
+'''
+Returns total price paid for individual rentals
+'''
+import argparse
+import json
+import datetime
+import math
+
+import logging
+import traceback
+import functools
+import sys
+
+def create_logger():
+    """
+    Creates a logging object and returns it, parse logging argument selective, by using decorators.
+    conditional logging with single command line variable which can turn logging on or off for decorated classes or functions.
+    """
+    logger = logging.getLogger()
+    logger.setLevel(logging.INFO)
+
+    log_format = "%(asctime)s %(filename)s:%(lineno)-3d %(levelname)s %(message)s"
+    formatter = logging.Formatter(log_format)
+    
+    # create the logging file handler, log filename format with timestamp 
+    file_handler = logging.FileHandler(datetime.datetime.now().strftime(" charges_calc %Y-%m-%d") + '.log')
+    file_handler.setFormatter(formatter)
+
+    # add handler to logger object
+    logger.addHandler(file_handler)
+
+    return logger
+
+
+def exception(function):
+    """
+    A decorator that wraps the passed in function and logs 
+    exceptions should one occur
+    """
+    @functools.wraps(function)
+    def wrapper(*args, **kwargs):
+        logger = create_logger()
+        try:
+            return function(*args, **kwargs)
+        except:
+            # log the exception
+            err = "There was an exception in  "
+            err += function.__name__
+            logger.exception(err)
+
+            # re-raise the exception
+            raise
+    return wrapper
+
+def parse_cmd_arguments():
+    parser = argparse.ArgumentParser(description='Process some integers and logging.')
+    parser.add_argument('-i', '--input', help='input JSON file', required=True)
+    parser.add_argument('-o', '--output', help='ouput JSON file', required=True)
+    # parser.add_argument('-d', required=True)
+    parser.add_argument('-d', '--debug', help='enable debug logging', required=False, default=0)
+
+    return parser.parse_args()
+
+@exception
+def load_rentals_file(filename):
+    with open(filename) as file:
+        try:
+            data = json.load(file)
+        except:
+            # Error loading json from file, for the reason expected property name error out.
+            logging.error("Error loading json from file {}".format(filename))
+            logging.warning("JSONDecodeError: Expecting property name enclosed in double quotes: line 5884 column 23")
+
+            # Here capture logging stacktrace or traceback to track the code’s execution error.
+            logging.exception("The next few lines are Traceback error message")
```

Comment:
You are not actually logging the traceback, and you are excepting all errors, so this is not true. What you want is something like this:
```
except json.JSONDecodeError as err:
    logging.error(err.message)
```
That will log the actual traceback, and only handles errors that you are prepared to handle.

Reply:
Thank you that is actually a good point! I just updated. I assumed logging.exception( ) will capture exception